### DISCUSSION TITLE
Annotation and setting font pointsize relative to image size

### DISCUSSION THREAD

**Discussion by trustieee** [ImageMagick/ImageMagick#5601] [Help] [answered]
I have images of varying size coming to me, they range from around 100x100 to 2000x2000.
I'm trying to find a way to place an annotation on some corner of the image, but to create extra white space for the annotation.

For example, with the following I can splice out a 20px height white background to place the annotation on, but obviously 20px doesn't scale well to a huge image. So I'd need a way to get a percentage of the original image's height.
```
magick convert jpg1.jpg -gravity Southwest -background White -splice 0x20 -annotate +0+2 "TEXT HERE" out.jpg
```

I'm ideally trying to do something like this to create a background image that's 5% taller than the original image:
```
create new background image with white background with dimensions
{original image's width, original image's height + (original image's height * .05)}
```
 and composite the original image on top of this new one. But I'm not sure how to incorporate the font size into this relative sizing, or how to translate this into ImageMagick commands.

**Comment by fmw42** [comment]
First in IM 7, use magick, not magick convert. So you want to pad all around or only on two sides or only on one side. If you want to pad all around you could extend the image by 10% of its size and centered so that it has 5% all around

`magick jpg1.jpg -background white -gravity center -extent "%[fx:1.1*w]x%[fx:1.1*h]" -annotate +0+2 "TEXT HERE" out.jpg`

If you just want to pad on the south and left, then 

`magick jpg1.jpg -background white -gravity southwest -extent "%[fx:1.05*w]x%[fx:1.05*h]" -annotate +0+2 "TEXT HERE" out.jpg`

**Comment by GeeMack** [comment, marked as answer]
First, you should not include "convert" in your command unless you actually want the behavior of ImageMagick version 6. Remove the "convert" and just use "magick" to use IMv7. That gives you the use of FX expressions in the command.

The following command reads the input image, uses its dimensions to set the size of a label to 25% of its width, creates the label, and composites it on the input image.

    magick jpg1.jpg -gravity southwest -background white -splice 0x5% -background none ^
       -bordercolor none -size %[fx:w*0.25]x ( label:"TEXT HERE" -border 5% ) -composite out.jpg

The `-splice` operation uses a percentage to add a strip to the bottom, and `-border` puts a border of a relative size around the label as padding when composited in the corner. Adjust those as needed.
 
To make the label size relative to the height instead of the width, use something like `-size x%[fx:h*0.05]`. Adjust the size of the border around the label to suit your need, even using one percentage for the left/right borders and another for the top/bottom, like `-border 2x5%`.

EDITED TO ADD: You can create that same result but without as much control with something as simple as this...

    magick jpg1.jpg -size %[w]x%[fx:h*0.05] label:"TEXT HERE" -append out2.jpg

Read the input, set the height for the text relative to the height of the image, create a label of that height, and append it to the bottom of the input image. The background for the label defaults to white. Use `-font Arial` to use your choice of fonts.

**Comment by snibgo** [comment]
The OP also asked about "`-pointsize N`". The units of pointsize is "points", where 1 point is 1/72 of an inch. So if you need a pointsize of, say, 20 pixels then you need to know the image density in dpi (dots per inch). Suppose the image dpi is 144. Then 20 pixels is 20*72/144 = 10 points, so you need "`-pointsize 10`".

EDIT: Corrected typo.
